### ISSUE TITLE
MSL: SubgroupSize becomes [[thread_execution_width]] instead of [[threads_per_simdgroup]] in compute shaders

### ISSUE THREAD

**Issue by Dredhog** [KhronosGroup/SPIRV-Cross#1386]
Hi, it seems that variables decorated with `BuiltIn SubgroupSize` end up as `[[thread_execution_width]]` but only in compute shaders, PS correctly produce `[[threads_per_simdgroup]]`

The input HLSL:
```
RWStructuredBuffer<float4> Result;

[numthreads(8,8,1)]
void main(uint3 id : SV_DispatchThreadID) : SV_Target
{
    Result[id.y].x = float(WaveGetLaneCount());
}
```
with `dxc.exe -E main -T cs_6_0 -spirv -fspv-target-env=vulkan1.1 repro_shader.txt -Fo repro_shader.spv` becomes:
```
; SPIR-V
; Version: 1.3
; Generator: Google spiregg; 0
; Bound: 25
; Schema: 0
               OpCapability Shader
               OpCapability GroupNonUniform
               OpMemoryModel Logical GLSL450
               OpEntryPoint GLCompute %main "main" %gl_GlobalInvocationID %SubgroupSize
               OpExecutionMode %main LocalSize 8 8 1
               OpSource HLSL 600
               OpName %type_RWStructuredBuffer_v4float "type.RWStructuredBuffer.v4float"
               OpName %Result "Result"
               OpName %main "main"
               OpDecorate %gl_GlobalInvocationID BuiltIn GlobalInvocationId
               OpDecorate %SubgroupSize BuiltIn SubgroupSize
               OpDecorate %Result DescriptorSet 0
               OpDecorate %Result Binding 0
               OpDecorate %_runtimearr_v4float ArrayStride 16
               OpMemberDecorate %type_RWStructuredBuffer_v4float 0 Offset 0
               OpDecorate %type_RWStructuredBuffer_v4float BufferBlock
       %uint = OpTypeInt 32 0
        %int = OpTypeInt 32 1
      %int_0 = OpConstant %int 0
      %float = OpTypeFloat 32
    %v4float = OpTypeVector %float 4
%_runtimearr_v4float = OpTypeRuntimeArray %v4float
%type_RWStructuredBuffer_v4float = OpTypeStruct %_runtimearr_v4float
%_ptr_Uniform_type_RWStructuredBuffer_v4float = OpTypePointer Uniform %type_RWStructuredBuffer_v4float
     %v3uint = OpTypeVector %uint 3
%_ptr_Input_v3uint = OpTypePointer Input %v3uint
%_ptr_Input_uint = OpTypePointer Input %uint
       %void = OpTypeVoid
         %17 = OpTypeFunction %void
%_ptr_Uniform_float = OpTypePointer Uniform %float
     %Result = OpVariable %_ptr_Uniform_type_RWStructuredBuffer_v4float Uniform
%gl_GlobalInvocationID = OpVariable %_ptr_Input_v3uint Input
%SubgroupSize = OpVariable %_ptr_Input_uint Input
       %main = OpFunction %void None %17
         %19 = OpLabel
         %20 = OpLoad %v3uint %gl_GlobalInvocationID
         %21 = OpLoad %uint %SubgroupSize
         %22 = OpConvertUToF %float %21
         %23 = OpCompositeExtract %uint %20 1
         %24 = OpAccessChain %_ptr_Uniform_float %Result %int_0 %23 %int_0
               OpStore %24 %22
               OpReturn
               OpFunctionEnd
```
Then `spirv-cross --msl --msl-version 20200 repro_shader.spv` outputs:
```
#include <metal_stdlib>
#include <simd/simd.h>

using namespace metal;

struct type_RWStructuredBuffer_v4float
{
    float4 _m0[1];
};

kernel void main0(device type_RWStructuredBuffer_v4float& Result [[buffer(0)]], uint3 gl_GlobalInvocationID [[thread_position_in_grid]], uint gl_SubgroupSize [[thread_execution_width]])
{
    Result._m0[gl_GlobalInvocationID.y].x = float(gl_SubgroupSize);
}
```
I also tried to generate the SPIR-V with `glslangValidator` but got the same end result (command used: `glslangValidator.exe -e main -D -S comp --target-env vulkan1.1 -V repro_shader.txt -o repro_shader.spv`)

**Comment by cdavis5e** [comment]
There is no actual semantic difference between `[[thread_execution_width]]` and `[[threads_per_simdgroup]]`; they're always the same. SPIRV-Cross uses the former in preference to the latter in compute shaders because the former is available since Metal 1.0.

**Comment by Dredhog** [comment]
Oh, ok, It seems I had mistaken [[thread_execution_width]] for the threadgroup size due to the presence of the dedicated [[threads_per_simdgroup]] attribute. Thanks for the clarification!
